cloud/pkg/controller/manager/location.go

```diff
@@ -113,18 +115,18 @@ func (lc *LocationCache) SecretNodes(namespace, name string) (nodes []string) {
 
 //IsEdgeNode checks weather node is edge node or not
 func (lc *LocationCache) IsEdgeNode(nodeName string) bool {
-	for _, node := range lc.edgeNodes {
+	for _, node := range lc.EdgeNodes {
 		if node == nodeName {
 			return true
 		}
 	}
 	return false
 }
 
-//UpdateEdgeNode is to maintain edge nodes name upto-date by querying kubernetes client
+// UpdateEdgeNode is to maintain edge nodes name upto-date by querying kubernetes client
 func (lc *LocationCache) UpdateEdgeNode(nodeName string) {
-	lc.edgeNodes = append(lc.edgeNodes, nodeName)
-	log.LOGGER.Infof("Edge nodes updated : %v \n", lc.edgeNodes)
+	lc.EdgeNodes = append(lc.EdgeNodes, nodeName)
```

Comment:
UpdateEdgeNode with two args: nodeName and state and the lc.EdgeNodes.Store(nodeName, state)

Reply:
used map instead of syncMap so saving directly